File: TS/系统/03．技能系统/00．技能模板+函数/04．机制组件/11．技能属性修正/01．施法距离修正.ts
/** @noSelfInFile */

const jass = require("jass.common") as any;
const platformAbilityAction = require("平台扩展API动作") as {
  技能_设置技能施法距离: (this: void, 单位: any, 技能代码: number, 值: number) => boolean;
};
const platformAbilityGetter = require("平台扩展API取值") as {
  技能_获取技能施法距离: (this: void, 单位: any, 技能代码: number) => number;
};
const { stringToFourCCSafe } = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版") as {
  stringToFourCCSafe: (this: void, s: string | undefined | null) => number;
};

const GetHandleId = jass.GetHandleId as (handle: any) => number;
const IsUnitType = jass.IsUnitType as (unit: any, unitType: any) => boolean;
const UNIT_TYPE_HERO = jass.UNIT_TYPE_HERO as any;

export type 标准技能距离修正用途 =
  | "施法距离"
  | "施法距离派生距离"
  | "自身位移距离"
  | "路径长度"
  | "路径总长度"
  | "直线长度"
  | "矩形长度"
  | "胶囊长度"
  | "扇形半径"
  | "弹幕飞行距离"
  | "飞行最大距离"
  | "投射物最大距离"
  | "追踪最大距离"
  | "目标点最大距离"
  | "落点最大距离"
  | "效果半径"
  | "爆炸半径"
  | "击退距离"
  | "牵引距离"
  | "移动别人距离";

export type 技能距离修正用途 = 标准技能距离修正用途 | (string & {});

export interface 英雄技能施法距离配置 {
  基础施法距离?: number;
  最小距离?: number;
  最大距离?: number;
  同步物遍施法距离?: boolean;
}

export interface 技能距离修正选项 {
  吃施法距离加成?: boolean;
  最小距离?: number;
  最大距离?: number;
}

const 英雄技能全局施法距离修正表: Record<number, number | undefined> = {};
const 英雄技能单技能施法距离修正表: Record<string, number | undefined> = {};
const 英雄技能施法距离配置表: Record<string, 英雄技能施法距离配置 | undefined> = {};

function 转技能ID(this: void, 技能ID: number | string | undefined | null): number {
  if (技能ID == null) return 0;
  if (typeof 技能ID === "number") return 技能ID;
  return stringToFourCCSafe(技能ID);
}

function 是有效英雄(this: void, 单位: any): boolean {
  return 单位 != null && 单位 !== 0 && IsUnitType(单位, UNIT_TYPE_HERO) === true;
}

function 取单位ID(this: void, 单位: any): number {
  if (单位 == null || 单位 === 0) return 0;
  return GetHandleId(单位) || 0;
}

function 取技能键(this: void, 单位: any, 技能ID: number | string | undefined | null): string {
  const 单位ID = 取单位ID(单位);
  const 实际技能ID = 转技能ID(技能ID);
  if (单位ID <= 0 || 实际技能ID <= 0) return "";
  return String(单位ID) + "#" + String(实际技能ID);
}

function 限制距离(this: void, 值: number, 最小距离?: number, 最大距离?: number): number {
  let result = 值;
  const min = 最小距离 ?? 0;
  if (result < min) result = min;
  if (最大距离 != null && 最大距离 > min && result > 最大距离) result = 最大距离;
  return result;
}

export function 技能距离用途默认吃施法距离加成(this: void, 用途: 技能距离修正用途): boolean {
  return 用途 === "施法距离"
    || 用途 === "施法距离派生距离"
    || 用途 === "自身位移距离"
    || 用途 === "路径长度"
    || 用途 === "路径总长度"
    || 用途 === "直线长度"
    || 用途 === "矩形长度"
    || 用途 === "胶囊长度"
    || 用途 === "扇形半径"
    || 用途 === "弹幕飞行距离"
    || 用途 === "飞行最大距离"
    || 用途 === "投射物最大距离"
    || 用途 === "追踪最大距离"
    || 用途 === "目标点最大距离"
    || 用途 === "落点最大距离";
}

export function 取英雄技能施法距离修正(this: void, 单位: any, 技能ID?: number | string): number {
  if (!是有效英雄(单位)) return 0;
  const 单位ID = 取单位ID(单位);
  let result = 英雄技能全局施法距离修正表[单位ID] ?? 0;
  const key = 技能ID != null ? 取技能键(单位, 技能ID) : "";
  if (key !== "") result += 英雄技能单技能施法距离修正表[key] ?? 0;
  return result;
}

export function 取英雄技能修正距离(this: void, 单位: any, 技能ID: number | string, 基础距离: number, 用途: 技能距离修正用途, 选项?: 技能距离修正选项): number {
  if (!(基础距离 > 0)) return 0;
  const shouldApply = 选项?.吃施法距离加成 ?? 技能距离用途默认吃施法距离加成(用途);
  if (!shouldApply) return 限制距离(基础距离, 选项?.最小距离, 选项?.最大距离);

  const key = 取技能键(单位, 技能ID);
  const config = key !== "" ? 英雄技能施法距离配置表[key] : undefined;
  const min = 选项?.最小距离 ?? config?.最小距离;
  const max = 选项?.最大距离 ?? config?.最大距离;
  return 限制距离(基础距离 + 取英雄技能施法距离修正(单位, 技能ID), min, max);
}

export function 取英雄技能施法距离派生距离(this: void, 单位: any, 技能ID: number | string, 基础距离: number, 选项?: 技能距离修正选项): number {
  return 取英雄技能修正距离(单位, 技能ID, 基础距离, "施法距离派生距离", {
    ...(选项 ?? {}),
    吃施法距离加成: 选项?.吃施法距离加成 ?? true,
  });
}

export function 取英雄技能最终施法距离(this: void, 单位: any, 技能ID: number | string, 基础施法距离?: number): number {
  const key = 取技能键(单位, 技能ID);
  const config = key !== "" ? 英雄技能施法距离配置表[key] : undefined;
  const abilityId = 转技能ID(技能ID);
  const base = 基础施法距离 ?? config?.基础施法距离 ?? (abilityId > 0 ? platformAbilityGetter.技能_获取技能施法距离(单位, abilityId) : 0);
  return 取英雄技能修正距离(单位, abilityId, base, "施法距离", {
    最小距离: config?.最小距离,
    最大距离: config?.最大距离,
  });
}

export function 登记英雄技能施法距离配置(this: void, 单位: any, 技能ID: number | string, 配置: 英雄技能施法距离配置): boolean {
  if (!是有效英雄(单位)) return false;
  const abilityId = 转技能ID(技能ID);
  const key = 取技能键(单位, abilityId);
  if (key === "") return false;

  const base = 配置.基础施法距离 ?? platformAbilityGetter.技能_获取技能施法距离(单位, abilityId);
  英雄技能施法距离配置表[key] = {
    ...配置,
    基础施法距离: base > 0 ? base : 配置.基础施法距离,
  };

  if (配置.同步物遍施法距离 !== false) {
    刷新英雄技能物遍施法距离(单位, abilityId);
  }
  return true;
}

export function 取消登记英雄技能施法距离配置(this: void, 单位: any, 技能ID: number | string): void {
  const key = 取技能键(单位, 技能ID);
  if (key === "") return;
  delete 英雄技能施法距离配置表[key];
  delete 英雄技能单技能施法距离修正表[key];
}

export function 设置英雄技能施法距离修正(this: void, 单位: any, 修正值: number, 刷新: boolean = true): void {
  if (!是有效英雄(单位)) return;
  const 单位ID = 取单位ID(单位);
  英雄技能全局施法距离修正表[单位ID] = 修正值;
  if (刷新) 刷新英雄单位全部技能施法距离(单位);
}

export function 调整英雄技能施法距离修正(this: void, 单位: any, 变化值: number, 刷新: boolean = true): void {
  设置英雄技能施法距离修正(单位, 取英雄技能施法距离修正(单位) + 变化值, 刷新);
}

export function 设置指定英雄技能施法距离修正(this: void, 单位: any, 技能ID: number | string, 修正值: number, 刷新: boolean = true): void {
  if (!是有效英雄(单位)) return;
  const key = 取技能键(单位, 技能ID);
  if (key === "") return;
  英雄技能单技能施法距离修正表[key] = 修正值;
  if (刷新) 刷新英雄技能物遍施法距离(单位, 技能ID);
}

export function 刷新英雄技能物遍施法距离(this: void, 单位: any, 技能ID: number | string, 基础施法距离?: number): boolean {
  if (!是有效英雄(单位)) return false;
  const abilityId = 转技能ID(技能ID);
  if (abilityId <= 0) return false;

  const key = 取技能键(单位, abilityId);
  const config = key !== "" ? 英雄技能施法距离配置表[key] : undefined;
  const base = 基础施法距离 ?? config?.基础施法距离;
  if (!(base != null && base > 0)) return false;

  const range = 取英雄技能最终施法距离(单位, abilityId, base);
  return platformAbilityAction.技能_设置技能施法距离(单位, abilityId, range);
}

export function 刷新英雄单位全部技能施法距离(this: void, 单位: any): void {
  if (!是有效英雄(单位)) return;
  const prefix = String(取单位ID(单位)) + "#";
  for (const key in 英雄技能施法距离配置表) {
    if (key.indexOf(prefix) !== 0) continue;
    const config = 英雄技能施法距离配置表[key];
    if (config == null || config.同步物遍施法距离 === false) continue;
    const abilityId = parseInt(key.substring(prefix.length), 10);
    if (abilityId > 0) 刷新英雄技能物遍施法距离(单位, abilityId);
  }
}

export function 清除英雄技能施法距离修正(this: void, 单位: any, 刷新: boolean = true): void {
  const 单位ID = 取单位ID(单位);
  if (单位ID <= 0) return;
  delete 英雄技能全局施法距离修正表[单位ID];
  const prefix = String(单位ID) + "#";
  for (const key in 英雄技能单技能施法距离修正表) {
    if (key.indexOf(prefix) === 0) delete 英雄技能单技能施法距离修正表[key];
  }
  if (刷新) 刷新英雄单位全部技能施法距离(单位);
}

export {};

File: src/系统/03．技能系统/00．技能模板+函数/04．机制组件/11．技能属性修正/01．施法距离修正.lua
local ____lualib = require("lualib_bundle")
local __TS__ObjectAssign = ____lualib.__TS__ObjectAssign
local __TS__SparseArrayNew = ____lualib.__TS__SparseArrayNew
local __TS__SparseArrayPush = ____lualib.__TS__SparseArrayPush
local __TS__SparseArraySpread = ____lualib.__TS__SparseArraySpread
local __TS__Delete = ____lualib.__TS__Delete
local __TS__StringSubstring = ____lualib.__TS__StringSubstring
local __TS__ParseInt = ____lualib.__TS__ParseInt
local ____exports = {}
local _____8F6C_6280_80FDID, _____662F_6709_6548_82F1_96C4, _____53D6_5355_4F4DID, _____53D6_6280_80FD_952E, _____9650_5236_8DDD_79BB, platformAbilityAction, platformAbilityGetter, stringToFourCCSafe, GetHandleId, IsUnitType, UNIT_TYPE_HERO, _____82F1_96C4_6280_80FD_5168_5C40_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868, _____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868, _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868
function _____8F6C_6280_80FDID(_____6280_80FDID)
    if _____6280_80FDID == nil then
        return 0
    end
    if type(_____6280_80FDID) == "number" then
        return _____6280_80FDID
    end
    return stringToFourCCSafe(_____6280_80FDID)
end
function _____662F_6709_6548_82F1_96C4(_____5355_4F4D)
    return _____5355_4F4D ~= nil and _____5355_4F4D ~= 0 and IsUnitType(_____5355_4F4D, UNIT_TYPE_HERO) == true
end
function _____53D6_5355_4F4DID(_____5355_4F4D)
    if _____5355_4F4D == nil or _____5355_4F4D == 0 then
        return 0
    end
    return GetHandleId(_____5355_4F4D) or 0
end
function _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID)
    local _____5355_4F4DID = _____53D6_5355_4F4DID(_____5355_4F4D)
    local _____5B9E_9645_6280_80FDID = _____8F6C_6280_80FDID(_____6280_80FDID)
    if _____5355_4F4DID <= 0 or _____5B9E_9645_6280_80FDID <= 0 then
        return ""
    end
    return (tostring(_____5355_4F4DID) .. "#") .. tostring(_____5B9E_9645_6280_80FDID)
end
function _____9650_5236_8DDD_79BB(_____503C, _____6700_5C0F_8DDD_79BB, _____6700_5927_8DDD_79BB)
    local result = _____503C
    local min = _____6700_5C0F_8DDD_79BB or 0
    if result < min then
        result = min
    end
    if _____6700_5927_8DDD_79BB ~= nil and _____6700_5927_8DDD_79BB > min and result > _____6700_5927_8DDD_79BB then
        result = _____6700_5927_8DDD_79BB
    end
    return result
end
____exports["技能距离用途默认吃施法距离加成"] = function(_____7528_9014)
    return _____7528_9014 == "施法距离" or _____7528_9014 == "施法距离派生距离" or _____7528_9014 == "自身位移距离" or _____7528_9014 == "路径长度" or _____7528_9014 == "路径总长度" or _____7528_9014 == "直线长度" or _____7528_9014 == "矩形长度" or _____7528_9014 == "胶囊长度" or _____7528_9014 == "扇形半径" or _____7528_9014 == "弹幕飞行距离" or _____7528_9014 == "飞行最大距离" or _____7528_9014 == "投射物最大距离" or _____7528_9014 == "追踪最大距离" or _____7528_9014 == "目标点最大距离" or _____7528_9014 == "落点最大距离"
end
____exports["取英雄技能施法距离修正"] = function(_____5355_4F4D, _____6280_80FDID)
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return 0
    end
    local _____5355_4F4DID = _____53D6_5355_4F4DID(_____5355_4F4D)
    local result = _____82F1_96C4_6280_80FD_5168_5C40_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868[_____5355_4F4DID] or 0
    local key = _____6280_80FDID ~= nil and _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID) or ""
    if key ~= "" then
        result = result + (_____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868[key] or 0)
    end
    return result
end
____exports["取英雄技能修正距离"] = function(_____5355_4F4D, _____6280_80FDID, _____57FA_7840_8DDD_79BB, _____7528_9014, _____9009_9879)
    if not (_____57FA_7840_8DDD_79BB > 0) then
        return 0
    end
    local ____temp_3 = _____9009_9879 and _____9009_9879["吃施法距离加成"]
    if ____temp_3 == nil then
        ____temp_3 = ____exports["技能距离用途默认吃施法距离加成"](_____7528_9014)
    end
    local shouldApply = ____temp_3
    if not shouldApply then
        return _____9650_5236_8DDD_79BB(_____57FA_7840_8DDD_79BB, _____9009_9879 and _____9009_9879["最小距离"], _____9009_9879 and _____9009_9879["最大距离"])
    end
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID)
    local ____temp_8
    if key ~= "" then
        ____temp_8 = _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868[key]
    else
        ____temp_8 = nil
    end
    local config = ____temp_8
    local min = _____9009_9879 and _____9009_9879["最小距离"] or config and config["最小距离"]
    local max = _____9009_9879 and _____9009_9879["最大距离"] or config and config["最大距离"]
    return _____9650_5236_8DDD_79BB(
        _____57FA_7840_8DDD_79BB + ____exports["取英雄技能施法距离修正"](_____5355_4F4D, _____6280_80FDID),
        min,
        max
    )
end
____exports["取英雄技能最终施法距离"] = function(_____5355_4F4D, _____6280_80FDID, _____57FA_7840_65BD_6CD5_8DDD_79BB)
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID)
    local ____temp_23
    if key ~= "" then
        ____temp_23 = _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868[key]
    else
        ____temp_23 = nil
    end
    local config = ____temp_23
    local abilityId = _____8F6C_6280_80FDID(_____6280_80FDID)
    local base = _____57FA_7840_65BD_6CD5_8DDD_79BB or config and config["基础施法距离"] or (abilityId > 0 and platformAbilityGetter["技能_获取技能施法距离"](_____5355_4F4D, abilityId) or 0)
    return ____exports["取英雄技能修正距离"](
        _____5355_4F4D,
        abilityId,
        base,
        "施法距离",
        {["最小距离"] = config and config["最小距离"], ["最大距离"] = config and config["最大距离"]}
    )
end
____exports["刷新英雄技能物遍施法距离"] = function(_____5355_4F4D, _____6280_80FDID, _____57FA_7840_65BD_6CD5_8DDD_79BB)
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return false
    end
    local abilityId = _____8F6C_6280_80FDID(_____6280_80FDID)
    if abilityId <= 0 then
        return false
    end
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, abilityId)
    local ____temp_30
    if key ~= "" then
        ____temp_30 = _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868[key]
    else
        ____temp_30 = nil
    end
    local config = ____temp_30
    local base = _____57FA_7840_65BD_6CD5_8DDD_79BB or config and config["基础施法距离"]
    if not (base ~= nil and base > 0) then
        return false
    end
    local range = ____exports["取英雄技能最终施法距离"](_____5355_4F4D, abilityId, base)
    return platformAbilityAction["技能_设置技能施法距离"](_____5355_4F4D, abilityId, range)
end
____exports["刷新英雄单位全部技能施法距离"] = function(_____5355_4F4D)
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return
    end
    local prefix = tostring(_____53D6_5355_4F4DID(_____5355_4F4D)) .. "#"
    for key in pairs(_____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868) do
        do
            if (string.find(key, prefix, nil, true) or 0) - 1 ~= 0 then
                goto __continue42
            end
            local config = _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868[key]
            if config == nil or config["同步物遍施法距离"] == false then
                goto __continue42
            end
            local abilityId = __TS__ParseInt(
                __TS__StringSubstring(key, #prefix),
                10
            )
            if abilityId > 0 then
                ____exports["刷新英雄技能物遍施法距离"](_____5355_4F4D, abilityId)
            end
        end
        ::__continue42::
    end
end
---
-- @noSelfInFile
local jass = require("jass.common")
platformAbilityAction = require("平台扩展API动作")
platformAbilityGetter = require("平台扩展API取值")
local ____require_result_0 = require("lib.扩展函数.封装函数.01．通用工具.01．FourCC转换安全版")
stringToFourCCSafe = ____require_result_0.stringToFourCCSafe
GetHandleId = jass.GetHandleId
IsUnitType = jass.IsUnitType
UNIT_TYPE_HERO = jass.UNIT_TYPE_HERO
_____82F1_96C4_6280_80FD_5168_5C40_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868 = {}
_____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868 = {}
_____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868 = {}
____exports["取英雄技能施法距离派生距离"] = function(_____5355_4F4D, _____6280_80FDID, _____57FA_7840_8DDD_79BB, _____9009_9879)
    local ____exports__53D6_82F1_96C4_6280_80FD_4FEE_6B63_8DDD_79BB_22 = ____exports["取英雄技能修正距离"]
    local ____array_21 = __TS__SparseArrayNew(_____5355_4F4D, _____6280_80FDID, _____57FA_7840_8DDD_79BB, "施法距离派生距离")
    local ____temp_20 = _____9009_9879 or ({})
    local ____temp_19 = _____9009_9879 and _____9009_9879["吃施法距离加成"]
    if ____temp_19 == nil then
        ____temp_19 = true
    end
    __TS__SparseArrayPush(
        ____array_21,
        __TS__ObjectAssign({}, ____temp_20, {["吃施法距离加成"] = ____temp_19})
    )
    return ____exports__53D6_82F1_96C4_6280_80FD_4FEE_6B63_8DDD_79BB_22(__TS__SparseArraySpread(____array_21))
end
____exports["登记英雄技能施法距离配置"] = function(_____5355_4F4D, _____6280_80FDID, _____914D_7F6E)
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return false
    end
    local abilityId = _____8F6C_6280_80FDID(_____6280_80FDID)
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, abilityId)
    if key == "" then
        return false
    end
    local base = _____914D_7F6E["基础施法距离"] or platformAbilityGetter["技能_获取技能施法距离"](_____5355_4F4D, abilityId)
    _____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868[key] = __TS__ObjectAssign({}, _____914D_7F6E, {["基础施法距离"] = base > 0 and base or _____914D_7F6E["基础施法距离"]})
    if _____914D_7F6E["同步物遍施法距离"] ~= false then
        ____exports["刷新英雄技能物遍施法距离"](_____5355_4F4D, abilityId)
    end
    return true
end
____exports["取消登记英雄技能施法距离配置"] = function(_____5355_4F4D, _____6280_80FDID)
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID)
    if key == "" then
        return
    end
    __TS__Delete(_____82F1_96C4_6280_80FD_65BD_6CD5_8DDD_79BB_914D_7F6E_8868, key)
    __TS__Delete(_____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868, key)
end
____exports["设置英雄技能施法距离修正"] = function(_____5355_4F4D, _____4FEE_6B63_503C, _____5237_65B0)
    if _____5237_65B0 == nil then
        _____5237_65B0 = true
    end
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return
    end
    local _____5355_4F4DID = _____53D6_5355_4F4DID(_____5355_4F4D)
    _____82F1_96C4_6280_80FD_5168_5C40_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868[_____5355_4F4DID] = _____4FEE_6B63_503C
    if _____5237_65B0 then
        ____exports["刷新英雄单位全部技能施法距离"](_____5355_4F4D)
    end
end
____exports["调整英雄技能施法距离修正"] = function(_____5355_4F4D, _____53D8_5316_503C, _____5237_65B0)
    if _____5237_65B0 == nil then
        _____5237_65B0 = true
    end
    ____exports["设置英雄技能施法距离修正"](
        _____5355_4F4D,
        ____exports["取英雄技能施法距离修正"](_____5355_4F4D) + _____53D8_5316_503C,
        _____5237_65B0
    )
end
____exports["设置指定英雄技能施法距离修正"] = function(_____5355_4F4D, _____6280_80FDID, _____4FEE_6B63_503C, _____5237_65B0)
    if _____5237_65B0 == nil then
        _____5237_65B0 = true
    end
    if not _____662F_6709_6548_82F1_96C4(_____5355_4F4D) then
        return
    end
    local key = _____53D6_6280_80FD_952E(_____5355_4F4D, _____6280_80FDID)
    if key == "" then
        return
    end
    _____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868[key] = _____4FEE_6B63_503C
    if _____5237_65B0 then
        ____exports["刷新英雄技能物遍施法距离"](_____5355_4F4D, _____6280_80FDID)
    end
end
____exports["清除英雄技能施法距离修正"] = function(_____5355_4F4D, _____5237_65B0)
    if _____5237_65B0 == nil then
        _____5237_65B0 = true
    end
    local _____5355_4F4DID = _____53D6_5355_4F4DID(_____5355_4F4D)
    if _____5355_4F4DID <= 0 then
        return
    end
    __TS__Delete(_____82F1_96C4_6280_80FD_5168_5C40_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868, _____5355_4F4DID)
    local prefix = tostring(_____5355_4F4DID) .. "#"
    for key in pairs(_____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868) do
        if (string.find(key, prefix, nil, true) or 0) - 1 == 0 then
            __TS__Delete(_____82F1_96C4_6280_80FD_5355_6280_80FD_65BD_6CD5_8DDD_79BB_4FEE_6B63_8868, key)
        end
    end
    if _____5237_65B0 then
        ____exports["刷新英雄单位全部技能施法距离"](_____5355_4F4D)
    end
end
return ____exports
